File: lesson_11_20241029/ht/elevator.js
function getClosestElevator(elevatorA, elevatorB, callFloor) {

    let distanceA = Math.abs(elevatorA - callFloor);
    let distanceB = Math.abs(elevatorB - callFloor);

    if (distanceA < distanceB) {
        return "A";
    } else if (distanceB < distanceA) {
        return "B";
    } else {
        return "A";
    }
}

let elevatorA = 0;
let elevatorB = 8;
let callFloor = 5; //вызываем с этажа

let chosenElevator = getClosestElevator(elevatorA, elevatorB, callFloor);

console.log("Отправляем лифт: " + chosenElevator);

File: lesson_11_20241029/ht/Elevator.java
import java.util.Scanner;

public class Elevator {
      public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int totalFloors = 19;

        int elevatorA = 0;
        int elevatorB = 8;

        System.out.println("Введите этаж вызова: ");
        int callFloor = scanner.nextInt();

        String chosenElevator = getClosestElevator(elevatorA, elevatorB, callFloor);

        System.out.println("Отправляем лифт: " + chosenElevator);
    }

    public static String getClosestElevator(int elevatorA, int elevatorB, int callFloor) {

        int distanceA = Math.abs(elevatorA - callFloor);
        int distanceB = Math.abs(elevatorB - callFloor);

        if (distanceA < distanceB) {
            return "A";
        } else if (distanceB < distanceA) {
            return "B";
        } else {
            return "A";
        }
    }

}
